### ISSUE TITLE
Question about paper

### ISSUE THREAD

**Issue by abeyang00** [mrharicot/monodepth#229]
What is dense correspondence field exactly? There seems to be no explanation about this in the paper. And when i read your code, it seems like disparity is regarded as correspondence field??

Thank you in advance

**Comment by macaodha** [comment]
Yes. For each pixel we predict a single scalar value specifying the disparity (i.e. scaled inverse depth). The "ground truth" disparity would point to the corresponding pixel in the second stereo view.

**Comment by zhongcl-thu** [comment]
@macaodha Does the 'dl' mean the disparity of the right image in the equation(3)? Because in the paper, you indicate that the reconstructed right image(~Ir) can be obtained from left image(Il) and the disparity image(dr).  For the disparity smoothness loss, you weight this cost with an edge-aware term using the image gradients. It means that the origin image  should be correspondent with the disparity image in the Eq.3. But, according to the Eq.3, the 'Il ' isn't correspondent with the 'dl'. 
I would really appreciate it if you could answer my question.
![image](https://user-images.githubusercontent.com/37363660/65234978-437d0980-db08-11e9-8e10-fae8e210384c.png)
![image](https://user-images.githubusercontent.com/37363660/65234989-4b3cae00-db08-11e9-961b-c9e4fadb655f.png)

**Comment by macaodha** [comment]
dl is aligned with the left image and dr with the right. We use backward mapping (B in the attached image) to reconstruct the right image from pixels in the left guided by the predicted right disparity. 

![image](https://user-images.githubusercontent.com/6660523/65262488-4e5f7c00-dabf-11e9-87e6-3d7ccf564c52.png)
